Crash when current weather is snow on some devices

### Steps to reproduce

1. update to 6.0.12
2. try to open
3. ???

### Expected behavior

the app should open

### Actual behavior

it doesn't open, Breezy Weather keeps stopping

### Weather source used

Open-Meteo

### Breezy Weather version

6.0.12

### Android version

LineageOS 20 GSI

### Device

Unihertz Jelly Star

### Other details

same on freenet and standard
I can open settings from app info

### Acknowledgements

- [x] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open OR closed issue.
- [x] I checked that my issue is not covered in the [Frequently Asked Questions](https://github.com/breezy-weather/breezy-weather/blob/main/HELP.md) document.
- [x] I have updated the app to **[version 6.0.12 or later](https://github.com/breezy-weather/breezy-weather/releases)** and I can reproduce the issue in this version.
- [x] I wrote my request in English, and any screenshot I attached are also in English (language can be changed from Settings > Appearance), unless my issue only happens with a specific language. I can write in my native language under the English text, if necessary.
- [x] I understand that this app is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY, so I may not get a positive reply to my request, if any at all.

today the app opens normally
force stopping didn't help but a reboot fixed it

```
--------- beginning of crash
11-23 00:46:26.346   964   964 E AndroidRuntime: FATAL EXCEPTION: main
11-23 00:46:26.346   964   964 E AndroidRuntime: Process: org.breezyweather, PID: 964
11-23 00:46:26.346   964   964 E AndroidRuntime: java.lang.RuntimeException: Unable to resume activity {org.breezyweather/org.breezyweather.ui.main.MainActivity}: java.lang.IllegalArgumentException: bound must be positive
11-23 00:46:26.346   964   964 E AndroidRuntime: 	at android.app.ActivityThread.performResumeActivity(ActivityThread.java:4774)
11-23 00:46:26.346   964   964 E AndroidRuntime: 	at android.app.ActivityThread.handleResumeActivity(ActivityThread.java:4807)
11-23 00:46:26.346   964   964 E AndroidRuntime: 	at android.app.servertransaction.ResumeActivityItem.execute(ResumeActivityItem.java:57)
11-23 00:46:26.346   964   964 E AndroidRuntime: 	at android.app.servertransaction.ActivityTransactionItem.execute(ActivityTransactionItem.java:45)
11-23 00:46:26.346   964   964 E AndroidRuntime: 	at android.app.servertransaction.TransactionExecutor.executeLifecycleState(TransactionExecutor.java:179)
11-23 00:46:26.346   964   964 E AndroidRuntime: 	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:97)
11-23 00:46:26.346   964   964 E AndroidRuntime: 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2307)
11-23 00:46:26.346   964   964 E AndroidRuntime: 	at android.os.Handler.dispatchMessage(Handler.java:106)
11-23 00:46:26.346   964   964 E AndroidRuntime: 	at android.os.Looper.loopOnce(Looper.java:201)
11-23 00:46:26.346   964   964 E AndroidRuntime: 	at android.os.Looper.loop(Looper.java:288)
11-23 00:46:26.346   964   964 E AndroidRuntime: 	at android.app.ActivityThread.main(ActivityThread.java:7924)
11-23 00:46:26.346   964   964 E AndroidRuntime: 	at java.lang.reflect.Method.invoke(Native Method)
11-23 00:46:26.346   964   964 E AndroidRuntime: 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548)
11-23 00:46:26.346   964   964 E AndroidRuntime: 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:936)
11-23 00:46:26.346   964   964 E AndroidRuntime: Caused by: java.lang.IllegalArgumentException: bound must be positive
11-23 00:46:26.346   964   964 E AndroidRuntime: 	at java.util.Random.nextInt(Random.java:388)
11-23 00:46:26.346   964   964 E AndroidRuntime: 	at o8.l.b(SourceFile:39)
11-23 00:46:26.346   964   964 E AndroidRuntime: 	at o8.l.<init>(SourceFile:87)
11-23 00:46:26.346   964   964 E AndroidRuntime: 	at o8.f.<init>(SourceFile:366)
11-23 00:46:26.346   964   964 E AndroidRuntime: 	at androidx.concurrent.futures.o.P(SourceFile:102)
11-23 00:46:26.346   964   964 E AndroidRuntime: 	at n8.g.c(SourceFile:21)
11-23 00:46:26.346   964   964 E AndroidRuntime: 	at n8.g.setDrawable(SourceFile:53)
11-23 00:46:26.346   964   964 E AndroidRuntime: 	at n8.i.setDrawable(SourceFile:12)
11-23 00:46:26.346   964   964 E AndroidRuntime: 	at org.breezyweather.ui.main.fragments.HomeFragment.y(SourceFile:13)
11-23 00:46:26.346   964   964 E AndroidRuntime: 	at h1.U.n(SourceFile:93)
11-23 00:46:26.346   964   964 E AndroidRuntime: 	at h1.U.k(SourceFile:43)
11-23 00:46:26.346   964   964 E AndroidRuntime: 	at h1.L.L(SourceFile:62)
11-23 00:46:26.346   964   964 E AndroidRuntime: 	at h1.L.u(SourceFile:36)
11-23 00:46:26.346   964   964 E AndroidRuntime: 	at h1.x.onPostResume(SourceFile:29)
11-23 00:46:26.346   964   964 E AndroidRuntime: 	at h.m.onPostResume(SourceFile:1)
11-23 00:46:26.346   964   964 E AndroidRuntime: 	at android.app.Activity.performResume(Activity.java:8499)
11-23 00:46:26.346   964   964 E AndroidRuntime: 	at android.app.ActivityThread.performResumeActivity(ActivityThread.java:4764)
11-23 00:46:26.346   964   964 E AndroidRuntime: 	... 13 more
11-23 00:46:27.750  1034  1034 E AndroidRuntime: FATAL EXCEPTION: main
11-23 00:46:27.750  1034  1034 E AndroidRuntime: Process: org.breezyweather, PID: 1034
11-23 00:46:27.750  1034  1034 E AndroidRuntime: java.lang.RuntimeException: Unable to resume activity {org.breezyweather/org.breezyweather.ui.main.MainActivity}: java.lang.IllegalArgumentException: bound must be positive
11-23 00:46:27.750  1034  1034 E AndroidRuntime: 	at android.app.ActivityThread.performResumeActivity(ActivityThread.java:4774)
11-23 00:46:27.750  1034  1034 E AndroidRuntime: 	at android.app.ActivityThread.handleResumeActivity(ActivityThread.java:4807)
11-23 00:46:27.750  1034  1034 E AndroidRuntime: 	at android.app.servertransaction.ResumeActivityItem.execute(ResumeActivityItem.java:57)
11-23 00:46:27.750  1034  1034 E AndroidRuntime: 	at android.app.servertransaction.ActivityTransactionItem.execute(ActivityTransactionItem.java:45)
11-23 00:46:27.750  1034  1034 E AndroidRuntime: 	at android.app.servertransaction.TransactionExecutor.executeLifecycleState(TransactionExecutor.java:179)
11-23 00:46:27.750  1034  1034 E AndroidRuntime: 	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:97)
11-23 00:46:27.750  1034  1034 E AndroidRuntime: 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2307)
11-23 00:46:27.750  1034  1034 E AndroidRuntime: 	at android.os.Handler.dispatchMessage(Handler.java:106)
11-23 00:46:27.750  1034  1034 E AndroidRuntime: 	at android.os.Looper.loopOnce(Looper.java:201)
11-23 00:46:27.750  1034  1034 E AndroidRuntime: 	at android.os.Looper.loop(Looper.java:288)
11-23 00:46:27.750  1034  1034 E AndroidRuntime: 	at android.app.ActivityThread.main(ActivityThread.java:7924)
11-23 00:46:27.750  1034  1034 E AndroidRuntime: 	at java.lang.reflect.Method.invoke(Native Method)
11-23 00:46:27.750  1034  1034 E AndroidRuntime: 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548)
11-23 00:46:27.750  1034  1034 E AndroidRuntime: 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:936)
11-23 00:46:27.750  1034  1034 E AndroidRuntime: Caused by: java.lang.IllegalArgumentException: bound must be positive
11-23 00:46:27.750  1034  1034 E AndroidRuntime: 	at java.util.Random.nextInt(Random.java:388)
11-23 00:46:27.750  1034  1034 E AndroidRuntime: 	at o8.l.b(SourceFile:39)
11-23 00:46:27.750  1034  1034 E AndroidRuntime: 	at o8.l.<init>(SourceFile:87)
11-23 00:46:27.750  1034  1034 E AndroidRuntime: 	at o8.f.<init>(SourceFile:366)
11-23 00:46:27.750  1034  1034 E AndroidRuntime: 	at androidx.concurrent.futures.o.P(SourceFile:102)
11-23 00:46:27.750  1034  1034 E AndroidRuntime: 	at n8.g.c(SourceFile:21)
11-23 00:46:27.750  1034  1034 E AndroidRuntime: 	at n8.g.setDrawable(SourceFile:53)
11-23 00:46:27.750  1034  1034 E AndroidRuntime: 	at n8.i.setDrawable(SourceFile:12)
11-23 00:46:27.750  1034  1034 E AndroidRuntime: 	at org.breezyweather.ui.main.fragments.HomeFragment.y(SourceFile:13)
11-23 00:46:27.750  1034  1034 E AndroidRuntime: 	at h1.U.n(SourceFile:93)
11-23 00:46:27.750  1034  1034 E AndroidRuntime: 	at h1.U.k(SourceFile:43)
11-23 00:46:27.750  1034  1034 E AndroidRuntime: 	at h1.L.L(SourceFile:62)
11-23 00:46:27.750  1034  1034 E AndroidRuntime: 	at h1.L.u(SourceFile:36)
11-23 00:46:27.750  1034  1034 E AndroidRuntime: 	at h1.x.onPostResume(SourceFile:29)
11-23 00:46:27.750  1034  1034 E AndroidRuntime: 	at h.m.onPostResume(SourceFile:1)
11-23 00:46:27.750  1034  1034 E AndroidRuntime: 	at android.app.Activity.performResume(Activity.java:8499)
11-23 00:46:27.750  1034  1034 E AndroidRuntime: 	at android.app.ActivityThread.performResumeActivity(ActivityThread.java:4764)
11-23 00:46:27.750  1034  1034 E AndroidRuntime: 	... 13 more
11-23 00:46:36.491  1104  1104 E AndroidRuntime: FATAL EXCEPTION: main
11-23 00:46:36.491  1104  1104 E AndroidRuntime: Process: org.breezyweather, PID: 1104
11-23 00:46:36.491  1104  1104 E AndroidRuntime: java.lang.RuntimeException: Unable to resume activity {org.breezyweather/org.breezyweather.ui.main.MainActivity}: java.lang.IllegalArgumentException: bound must be positive
11-23 00:46:36.491  1104  1104 E AndroidRuntime: 	at android.app.ActivityThread.performResumeActivity(ActivityThread.java:4774)
11-23 00:46:36.491  1104  1104 E AndroidRuntime: 	at android.app.ActivityThread.handleResumeActivity(ActivityThread.java:4807)
11-23 00:46:36.491  1104  1104 E AndroidRuntime: 	at android.app.servertransaction.ResumeActivityItem.execute(ResumeActivityItem.java:57)
11-23 00:46:36.491  1104  1104 E AndroidRuntime: 	at android.app.servertransaction.ActivityTransactionItem.execute(ActivityTransactionItem.java:45)
11-23 00:46:36.491  1104  1104 E AndroidRuntime: 	at android.app.servertransaction.TransactionExecutor.executeLifecycleState(TransactionExecutor.java:179)
11-23 00:46:36.491  1104  1104 E AndroidRuntime: 	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:97)
11-23 00:46:36.491  1104  1104 E AndroidRuntime: 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2307)
11-23 00:46:36.491  1104  1104 E AndroidRuntime: 	at android.os.Handler.dispatchMessage(Handler.java:106)
11-23 00:46:36.491  1104  1104 E AndroidRuntime: 	at android.os.Looper.loopOnce(Looper.java:201)
11-23 00:46:36.491  1104  1104 E AndroidRuntime: 	at android.os.Looper.loop(Looper.java:288)
11-23 00:46:36.491  1104  1104 E AndroidRuntime: 	at android.app.ActivityThread.main(ActivityThread.java:7924)
11-23 00:46:36.491  1104  1104 E AndroidRuntime: 	at java.lang.reflect.Method.invoke(Native Method)
11-23 00:46:36.491  1104  1104 E AndroidRuntime: 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548)
11-23 00:46:36.491  1104  1104 E AndroidRuntime: 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:936)
11-23 00:46:36.491  1104  1104 E AndroidRuntime: Caused by: java.lang.IllegalArgumentException: bound must be positive
11-23 00:46:36.491  1104  1104 E AndroidRuntime: 	at java.util.Random.nextInt(Random.java:388)
11-23 00:46:36.491  1104  1104 E AndroidRuntime: 	at o8.l.b(SourceFile:39)
11-23 00:46:36.491  1104  1104 E AndroidRuntime: 	at o8.l.<init>(SourceFile:87)
11-23 00:46:36.491  1104  1104 E AndroidRuntime: 	at o8.f.<init>(SourceFile:366)
11-23 00:46:36.491  1104  1104 E AndroidRuntime: 	at androidx.concurrent.futures.o.P(SourceFile:102)
11-23 00:46:36.491  1104  1104 E AndroidRuntime: 	at n8.g.c(SourceFile:21)
11-23 00:46:36.491  1104  1104 E AndroidRuntime: 	at n8.g.setDrawable(SourceFile:53)
11-23 00:46:36.491  1104  1104 E AndroidRuntime: 	at n8.i.setDrawable(SourceFile:12)
11-23 00:46:36.491  1104  1104 E AndroidRuntime: 	at org.breezyweather.ui.main.fragments.HomeFragment.y(SourceFile:13)
11-23 00:46:36.491  1104  1104 E AndroidRuntime: 	at h1.U.n(SourceFile:93)
11-23 00:46:36.491  1104  1104 E AndroidRuntime: 	at h1.U.k(SourceFile:43)
11-23 00:46:36.491  1104  1104 E AndroidRuntime: 	at h1.L.L(SourceFile:62)
11-23 00:46:36.491  1104  1104 E AndroidRuntime: 	at h1.L.u(SourceFile:36)
11-23 00:46:36.491  1104  1104 E AndroidRuntime: 	at h1.x.onPostResume(SourceFile:29)
11-23 00:46:36.491  1104  1104 E AndroidRuntime: 	at h.m.onPostResume(SourceFile:1)
11-23 00:46:36.491  1104  1104 E AndroidRuntime: 	at android.app.Activity.performResume(Activity.java:8499)
11-23 00:46:36.491  1104  1104 E AndroidRuntime: 	at android.app.ActivityThread.performResumeActivity(ActivityThread.java:4764)
11-23 00:46:36.491  1104  1104 E AndroidRuntime: 	... 13 more
11-23 00:46:40.642  1240  1240 E AndroidRuntime: FATAL EXCEPTION: main
11-23 00:46:40.642  1240  1240 E AndroidRuntime: Process: org.breezyweather, PID: 1240
11-23 00:46:40.642  1240  1240 E AndroidRuntime: java.lang.RuntimeException: Unable to resume activity {org.breezyweather/org.breezyweather.ui.main.MainActivity}: java.lang.IllegalArgumentException: bound must be positive
11-23 00:46:40.642  1240  1240 E AndroidRuntime: 	at android.app.ActivityThread.performResumeActivity(ActivityThread.java:4774)
11-23 00:46:40.642  1240  1240 E AndroidRuntime: 	at android.app.ActivityThread.handleResumeActivity(ActivityThread.java:4807)
11-23 00:46:40.642  1240  1240 E AndroidRuntime: 	at android.app.servertransaction.ResumeActivityItem.execute(ResumeActivityItem.java:57)
11-23 00:46:40.642  1240  1240 E AndroidRuntime: 	at android.app.servertransaction.ActivityTransactionItem.execute(ActivityTransactionItem.java:45)
11-23 00:46:40.642  1240  1240 E AndroidRuntime: 	at android.app.servertransaction.TransactionExecutor.executeLifecycleState(TransactionExecutor.java:179)
11-23 00:46:40.642  1240  1240 E AndroidRuntime: 	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:97)
11-23 00:46:40.642  1240  1240 E AndroidRuntime: 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2307)
11-23 00:46:40.642  1240  1240 E AndroidRuntime: 	at android.os.Handler.dispatchMessage(Handler.java:106)
11-23 00:46:40.642  1240  1240 E AndroidRuntime: 	at android.os.Looper.loopOnce(Looper.java:201)
11-23 00:46:40.642  1240  1240 E AndroidRuntime: 	at android.os.Looper.loop(Looper.java:288)
11-23 00:46:40.642  1240  1240 E AndroidRuntime: 	at android.app.ActivityThread.main(ActivityThread.java:7924)
11-23 00:46:40.642  1240  1240 E AndroidRuntime: 	at java.lang.reflect.Method.invoke(Native Method)
11-23 00:46:40.642  1240  1240 E AndroidRuntime: 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548)
11-23 00:46:40.642  1240  1240 E AndroidRuntime: 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:936)
11-23 00:46:40.642  1240  1240 E AndroidRuntime: Caused by: java.lang.IllegalArgumentException: bound must be positive
11-23 00:46:40.642  1240  1240 E AndroidRuntime: 	at java.util.Random.nextInt(Random.java:388)
11-23 00:46:40.642  1240  1240 E AndroidRuntime: 	at o8.l.b(SourceFile:39)
11-23 00:46:40.642  1240  1240 E AndroidRuntime: 	at o8.l.<init>(SourceFile:87)
11-23 00:46:40.642  1240  1240 E AndroidRuntime: 	at o8.f.<init>(SourceFile:366)
11-23 00:46:40.642  1240  1240 E AndroidRuntime: 	at androidx.concurrent.futures.o.P(SourceFile:102)
11-23 00:46:40.642  1240  1240 E AndroidRuntime: 	at n8.g.c(SourceFile:21)
11-23 00:46:40.642  1240  1240 E AndroidRuntime: 	at n8.g.setDrawable(SourceFile:53)
11-23 00:46:40.642  1240  1240 E AndroidRuntime: 	at n8.i.setDrawable(SourceFile:12)
11-23 00:46:40.642  1240  1240 E AndroidRuntime: 	at org.breezyweather.ui.main.fragments.HomeFragment.y(SourceFile:13)
11-23 00:46:40.642  1240  1240 E AndroidRuntime: 	at h1.U.n(SourceFile:93)
11-23 00:46:40.642  1240  1240 E AndroidRuntime: 	at h1.U.k(SourceFile:43)
11-23 00:46:40.642  1240  1240 E AndroidRuntime: 	at h1.L.L(SourceFile:62)
11-23 00:46:40.642  1240  1240 E AndroidRuntime: 	at h1.L.u(SourceFile:36)
11-23 00:46:40.642  1240  1240 E AndroidRuntime: 	at h1.x.onPostResume(SourceFile:29)
11-23 00:46:40.642  1240  1240 E AndroidRuntime: 	at h.m.onPostResume(SourceFile:1)
11-23 00:46:40.642  1240  1240 E AndroidRuntime: 	at android.app.Activity.performResume(Activity.java:8499)
11-23 00:46:40.642  1240  1240 E AndroidRuntime: 	at android.app.ActivityThread.performResumeActivity(ActivityThread.java:4764)
11-23 00:46:40.642  1240  1240 E AndroidRuntime: 	... 13 more
11-23 00:47:30.404  2103  2103 E AndroidRuntime: FATAL EXCEPTION: main
11-23 00:47:30.404  2103  2103 E AndroidRuntime: Process: org.breezyweather, PID: 2103
11-23 00:47:30.404  2103  2103 E AndroidRuntime: java.lang.RuntimeException: Unable to resume activity {org.breezyweather/org.breezyweather.ui.main.MainActivity}: java.lang.IllegalArgumentException: bound must be positive
11-23 00:47:30.404  2103  2103 E AndroidRuntime: 	at android.app.ActivityThread.performResumeActivity(ActivityThread.java:4774)
11-23 00:47:30.404  2103  2103 E AndroidRuntime: 	at android.app.ActivityThread.handleResumeActivity(ActivityThread.java:4807)
11-23 00:47:30.404  2103  2103 E AndroidRuntime: 	at android.app.servertransaction.ResumeActivityItem.execute(ResumeActivityItem.java:57)
11-23 00:47:30.404  2103  2103 E AndroidRuntime: 	at android.app.servertransaction.ActivityTransactionItem.execute(ActivityTransactionItem.java:45)
11-23 00:47:30.404  2103  2103 E AndroidRuntime: 	at android.app.servertransaction.TransactionExecutor.executeLifecycleState(TransactionExecutor.java:179)
11-23 00:47:30.404  2103  2103 E AndroidRuntime: 	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:97)
11-23 00:47:30.404  2103  2103 E AndroidRuntime: 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2307)
11-23 00:47:30.404  2103  2103 E AndroidRuntime: 	at android.os.Handler.dispatchMessage(Handler.java:106)
11-23 00:47:30.404  2103  2103 E AndroidRuntime: 	at android.os.Looper.loopOnce(Looper.java:201)
11-23 00:47:30.404  2103  2103 E AndroidRuntime: 	at android.os.Looper.loop(Looper.java:288)
11-23 00:47:30.404  2103  2103 E AndroidRuntime: 	at android.app.ActivityThread.main(ActivityThread.java:7924)
11-23 00:47:30.404  2103  2103 E AndroidRuntime: 	at java.lang.reflect.Method.invoke(Native Method)
11-23 00:47:30.404  2103  2103 E AndroidRuntime: 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548)
11-23 00:47:30.404  2103  2103 E AndroidRuntime: 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:936)
11-23 00:47:30.404  2103  2103 E AndroidRuntime: Caused by: java.lang.IllegalArgumentException: bound must be positive
11-23 00:47:30.404  2103  2103 E AndroidRuntime: 	at java.util.Random.nextInt(Random.java:388)
11-23 00:47:30.404  2103  2103 E AndroidRuntime: 	at r7.l.b(SourceFile:39)
11-23 00:47:30.404  2103  2103 E AndroidRuntime: 	at r7.l.<init>(SourceFile:87)
11-23 00:47:30.404  2103  2103 E AndroidRuntime: 	at r7.f.<init>(SourceFile:368)
11-23 00:47:30.404  2103  2103 E AndroidRuntime: 	at androidx.work.K.v(SourceFile:102)
11-23 00:47:30.404  2103  2103 E AndroidRuntime: 	at q7.e.c(SourceFile:21)
11-23 00:47:30.404  2103  2103 E AndroidRuntime: 	at q7.e.setDrawable(SourceFile:53)
11-23 00:47:30.404  2103  2103 E AndroidRuntime: 	at q7.g.setDrawable(SourceFile:12)
11-23 00:47:30.404  2103  2103 E AndroidRuntime: 	at org.breezyweather.ui.main.fragments.HomeFragment.y(SourceFile:13)
11-23 00:47:30.404  2103  2103 E AndroidRuntime: 	at h1.U.n(SourceFile:93)
11-23 00:47:30.404  2103  2103 E AndroidRuntime: 	at h1.U.k(SourceFile:43)
11-23 00:47:30.404  2103  2103 E AndroidRuntime: 	at h1.L.L(SourceFile:62)
11-23 00:47:30.404  2103  2103 E AndroidRuntime: 	at h1.L.u(SourceFile:36)
11-23 00:47:30.404  2103  2103 E AndroidRuntime: 	at h1.x.onPostResume(SourceFile:29)
11-23 00:47:30.404  2103  2103 E AndroidRuntime: 	at h.l.onPostResume(SourceFile:1)
11-23 00:47:30.404  2103  2103 E AndroidRuntime: 	at android.app.Activity.performResume(Activity.java:8499)
11-23 00:47:30.404  2103  2103 E AndroidRuntime: 	at android.app.ActivityThread.performResumeActivity(ActivityThread.java:4764)
11-23 00:47:30.404  2103  2103 E AndroidRuntime: 	... 13 more
11-23 00:49:02.001  3031  3031 E AndroidRuntime: FATAL EXCEPTION: main
11-23 00:49:02.001  3031  3031 E AndroidRuntime: Process: org.breezyweather, PID: 3031
11-23 00:49:02.001  3031  3031 E AndroidRuntime: java.lang.RuntimeException: Unable to resume activity {org.breezyweather/org.breezyweather.ui.main.MainActivity}: java.lang.IllegalArgumentException: bound must be positive
11-23 00:49:02.001  3031  3031 E AndroidRuntime: 	at android.app.ActivityThread.performResumeActivity(ActivityThread.java:4774)
11-23 00:49:02.001  3031  3031 E AndroidRuntime: 	at android.app.ActivityThread.handleResumeActivity(ActivityThread.java:4807)
11-23 00:49:02.001  3031  3031 E AndroidRuntime: 	at android.app.servertransaction.ResumeActivityItem.execute(ResumeActivityItem.java:57)
11-23 00:49:02.001  3031  3031 E AndroidRuntime: 	at android.app.servertransaction.ActivityTransactionItem.execute(ActivityTransactionItem.java:45)
11-23 00:49:02.001  3031  3031 E AndroidRuntime: 	at android.app.servertransaction.TransactionExecutor.executeLifecycleState(TransactionExecutor.java:179)
11-23 00:49:02.001  3031  3031 E AndroidRuntime: 	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:97)
11-23 00:49:02.001  3031  3031 E AndroidRuntime: 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2307)
11-23 00:49:02.001  3031  3031 E AndroidRuntime: 	at android.os.Handler.dispatchMessage(Handler.java:106)
11-23 00:49:02.001  3031  3031 E AndroidRuntime: 	at android.os.Looper.loopOnce(Looper.java:201)
11-23 00:49:02.001  3031  3031 E AndroidRuntime: 	at android.os.Looper.loop(Looper.java:288)
11-23 00:49:02.001  3031  3031 E AndroidRuntime: 	at android.app.ActivityThread.main(ActivityThread.java:7924)
11-23 00:49:02.001  3031  3031 E AndroidRuntime: 	at java.lang.reflect.Method.invoke(Native Method)
11-23 00:49:02.001  3031  3031 E AndroidRuntime: 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548)
11-23 00:49:02.001  3031  3031 E AndroidRuntime: 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:936)
11-23 00:49:02.001  3031  3031 E AndroidRuntime: Caused by: java.lang.IllegalArgumentException: bound must be positive
11-23 00:49:02.001  3031  3031 E AndroidRuntime: 	at java.util.Random.nextInt(Random.java:388)
11-23 00:49:02.001  3031  3031 E AndroidRuntime: 	at o8.l.b(SourceFile:39)
11-23 00:49:02.001  3031  3031 E AndroidRuntime: 	at o8.l.<init>(SourceFile:87)
11-23 00:49:02.001  3031  3031 E AndroidRuntime: 	at o8.f.<init>(SourceFile:366)
11-23 00:49:02.001  3031  3031 E AndroidRuntime: 	at androidx.concurrent.futures.o.P(SourceFile:102)
11-23 00:49:02.001  3031  3031 E AndroidRuntime: 	at n8.g.c(SourceFile:21)
11-23 00:49:02.001  3031  3031 E AndroidRuntime: 	at n8.g.setDrawable(SourceFile:53)
11-23 00:49:02.001  3031  3031 E AndroidRuntime: 	at n8.i.setDrawable(SourceFile:12)
11-23 00:49:02.001  3031  3031 E AndroidRuntime: 	at org.breezyweather.ui.main.fragments.HomeFragment.y(SourceFile:13)
11-23 00:49:02.001  3031  3031 E AndroidRuntime: 	at h1.U.n(SourceFile:93)
11-23 00:49:02.001  3031  3031 E AndroidRuntime: 	at h1.U.k(SourceFile:43)
11-23 00:49:02.001  3031  3031 E AndroidRuntime: 	at h1.L.L(SourceFile:62)
11-23 00:49:02.001  3031  3031 E AndroidRuntime: 	at h1.L.u(SourceFile:36)
11-23 00:49:02.001  3031  3031 E AndroidRuntime: 	at h1.x.onPostResume(SourceFile:29)
11-23 00:49:02.001  3031  3031 E AndroidRuntime: 	at h.m.onPostResume(SourceFile:1)
11-23 00:49:02.001  3031  3031 E AndroidRuntime: 	at android.app.Activity.performResume(Activity.java:8499)
11-23 00:49:02.001  3031  3031 E AndroidRuntime: 	at android.app.ActivityThread.performResumeActivity(ActivityThread.java:4764)
11-23 00:49:02.001  3031  3031 E AndroidRuntime: 	... 13 more
11-23 00:57:04.093  4984  4984 E AndroidRuntime: FATAL EXCEPTION: main
11-23 00:57:04.093  4984  4984 E AndroidRuntime: Process: org.breezyweather, PID: 4984
11-23 00:57:04.093  4984  4984 E AndroidRuntime: java.lang.RuntimeException: Unable to resume activity {org.breezyweather/org.breezyweather.ui.main.MainActivity}: java.lang.IllegalArgumentException: bound must be positive
11-23 00:57:04.093  4984  4984 E AndroidRuntime: 	at android.app.ActivityThread.performResumeActivity(ActivityThread.java:4774)
11-23 00:57:04.093  4984  4984 E AndroidRuntime: 	at android.app.ActivityThread.handleResumeActivity(ActivityThread.java:4807)
11-23 00:57:04.093  4984  4984 E AndroidRuntime: 	at android.app.servertransaction.ResumeActivityItem.execute(ResumeActivityItem.java:57)
11-23 00:57:04.093  4984  4984 E AndroidRuntime: 	at android.app.servertransaction.ActivityTransactionItem.execute(ActivityTransactionItem.java:45)
11-23 00:57:04.093  4984  4984 E AndroidRuntime: 	at android.app.servertransaction.TransactionExecutor.executeLifecycleState(TransactionExecutor.java:179)
11-23 00:57:04.093  4984  4984 E AndroidRuntime: 	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:97)
11-23 00:57:04.093  4984  4984 E AndroidRuntime: 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2307)
11-23 00:57:04.093  4984  4984 E AndroidRuntime: 	at android.os.Handler.dispatchMessage(Handler.java:106)
11-23 00:57:04.093  4984  4984 E AndroidRuntime: 	at android.os.Looper.loopOnce(Looper.java:201)
11-23 00:57:04.093  4984  4984 E AndroidRuntime: 	at android.os.Looper.loop(Looper.java:288)
11-23 00:57:04.093  4984  4984 E AndroidRuntime: 	at android.app.ActivityThread.main(ActivityThread.java:7924)
11-23 00:57:04.093  4984  4984 E AndroidRuntime: 	at java.lang.reflect.Method.invoke(Native Method)
11-23 00:57:04.093  4984  4984 E AndroidRuntime: 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548)
11-23 00:57:04.093  4984  4984 E AndroidRuntime: 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:936)
11-23 00:57:04.093  4984  4984 E AndroidRuntime: Caused by: java.lang.IllegalArgumentException: bound must be positive
11-23 00:57:04.093  4984  4984 E AndroidRuntime: 	at java.util.Random.nextInt(Random.java:388)
11-23 00:57:04.093  4984  4984 E AndroidRuntime: 	at o8.l.b(SourceFile:39)
11-23 00:57:04.093  4984  4984 E AndroidRuntime: 	at o8.l.<init>(SourceFile:87)
11-23 00:57:04.093  4984  4984 E AndroidRuntime: 	at o8.f.<init>(SourceFile:366)
11-23 00:57:04.093  4984  4984 E AndroidRuntime: 	at androidx.concurrent.futures.o.P(SourceFile:102)
11-23 00:57:04.093  4984  4984 E AndroidRuntime: 	at n8.g.c(SourceFile:21)
11-23 00:57:04.093  4984  4984 E AndroidRuntime: 	at n8.g.setDrawable(SourceFile:53)
11-23 00:57:04.093  4984  4984 E AndroidRuntime: 	at n8.i.setDrawable(SourceFile:12)
11-23 00:57:04.093  4984  4984 E AndroidRuntime: 	at org.breezyweather.ui.main.fragments.HomeFragment.y(SourceFile:13)
11-23 00:57:04.093  4984  4984 E AndroidRuntime: 	at h1.U.n(SourceFile:93)
11-23 00:57:04.093  4984  4984 E AndroidRuntime: 	at h1.U.k(SourceFile:43)
11-23 00:57:04.093  4984  4984 E AndroidRuntime: 	at h1.L.L(SourceFile:62)
11-23 00:57:04.093  4984  4984 E AndroidRuntime: 	at h1.L.u(SourceFile:36)
11-23 00:57:04.093  4984  4984 E AndroidRuntime: 	at h1.x.onPostResume(SourceFile:29)
11-23 00:57:04.093  4984  4984 E AndroidRuntime: 	at h.m.onPostResume(SourceFile:1)
11-23 00:57:04.093  4984  4984 E AndroidRuntime: 	at android.app.Activity.performResume(Activity.java:8499)
11-23 00:57:04.093  4984  4984 E AndroidRuntime: 	at android.app.ActivityThread.performResumeActivity(ActivityThread.java:4764)
11-23 00:57:04.093  4984  4984 E AndroidRuntime: 	... 13 more
11-23 00:57:07.036  5051  5051 E AndroidRuntime: FATAL EXCEPTION: main
11-23 00:57:07.036  5051  5051 E AndroidRuntime: Process: org.breezyweather, PID: 5051
11-23 00:57:07.036  5051  5051 E AndroidRuntime: java.lang.RuntimeException: Unable to resume activity {org.breezyweather/org.breezyweather.ui.main.MainActivity}: java.lang.IllegalArgumentException: bound must be positive
11-23 00:57:07.036  5051  5051 E AndroidRuntime: 	at android.app.ActivityThread.performResumeActivity(ActivityThread.java:4774)
11-23 00:57:07.036  5051  5051 E AndroidRuntime: 	at android.app.ActivityThread.handleResumeActivity(ActivityThread.java:4807)
11-23 00:57:07.036  5051  5051 E AndroidRuntime: 	at android.app.servertransaction.ResumeActivityItem.execute(ResumeActivityItem.java:57)
11-23 00:57:07.036  5051  5051 E AndroidRuntime: 	at android.app.servertransaction.ActivityTransactionItem.execute(ActivityTransactionItem.java:45)
11-23 00:57:07.036  5051  5051 E AndroidRuntime: 	at android.app.servertransaction.TransactionExecutor.executeLifecycleState(TransactionExecutor.java:179)
11-23 00:57:07.036  5051  5051 E AndroidRuntime: 	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:97)
11-23 00:57:07.036  5051  5051 E AndroidRuntime: 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2307)
11-23 00:57:07.036  5051  5051 E AndroidRuntime: 	at android.os.Handler.dispatchMessage(Handler.java:106)
11-23 00:57:07.036  5051  5051 E AndroidRuntime: 	at android.os.Looper.loopOnce(Looper.java:201)
11-23 00:57:07.036  5051  5051 E AndroidRuntime: 	at android.os.Looper.loop(Looper.java:288)
11-23 00:57:07.036  5051  5051 E AndroidRuntime: 	at android.app.ActivityThread.main(ActivityThread.java:7924)
11-23 00:57:07.036  5051  5051 E AndroidRuntime: 	at java.lang.reflect.Method.invoke(Native Method)
11-23 00:57:07.036  5051  5051 E AndroidRuntime: 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548)
11-23 00:57:07.036  5051  5051 E AndroidRuntime: 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:936)
11-23 00:57:07.036  5051  5051 E AndroidRuntime: Caused by: java.lang.IllegalArgumentException: bound must be positive
11-23 00:57:07.036  5051  5051 E AndroidRuntime: 	at java.util.Random.nextInt(Random.java:388)
11-23 00:57:07.036  5051  5051 E AndroidRuntime: 	at o8.l.b(SourceFile:39)
11-23 00:57:07.036  5051  5051 E AndroidRuntime: 	at o8.l.<init>(SourceFile:87)
11-23 00:57:07.036  5051  5051 E AndroidRuntime: 	at o8.f.<init>(SourceFile:366)
11-23 00:57:07.036  5051  5051 E AndroidRuntime: 	at androidx.concurrent.futures.o.P(SourceFile:102)
11-23 00:57:07.036  5051  5051 E AndroidRuntime: 	at n8.g.c(SourceFile:21)
11-23 00:57:07.036  5051  5051 E AndroidRuntime: 	at n8.g.setDrawable(SourceFile:53)
11-23 00:57:07.036  5051  5051 E AndroidRuntime: 	at n8.i.setDrawable(SourceFile:12)
11-23 00:57:07.036  5051  5051 E AndroidRuntime: 	at org.breezyweather.ui.main.fragments.HomeFragment.y(SourceFile:13)
11-23 00:57:07.036  5051  5051 E AndroidRuntime: 	at h1.U.n(SourceFile:93)
11-23 00:57:07.036  5051  5051 E AndroidRuntime: 	at h1.U.k(SourceFile:43)
11-23 00:57:07.036  5051  5051 E AndroidRuntime: 	at h1.L.L(SourceFile:62)
11-23 00:57:07.036  5051  5051 E AndroidRuntime: 	at h1.L.u(SourceFile:36)
11-23 00:57:07.036  5051  5051 E AndroidRuntime: 	at h1.x.onPostResume(SourceFile:29)
11-23 00:57:07.036  5051  5051 E AndroidRuntime: 	at h.m.onPostResume(SourceFile:1)
11-23 00:57:07.036  5051  5051 E AndroidRuntime: 	at android.app.Activity.performResume(Activity.java:8499)
11-23 00:57:07.036  5051  5051 E AndroidRuntime: 	at android.app.ActivityThread.performResumeActivity(ActivityThread.java:4764)
11-23 00:57:07.036  5051  5051 E AndroidRuntime: 	... 13 more
--------- beginning of main
11-23 01:00:25.207  5176  5199 E OpenGLRenderer: Unable to match the desired swap behavior.
App version: 6.0.12 (basic, e9329e0d, 60012
Android version: 13 (SDK 33); build lineage_arm64_bvN-userdebug 13 TQ3A.230901.001 eng.crossg.20251022.000945 test-keys
Device brand: Unihertz
Device manufacturer: Unihertz
Device name: Jelly_Star (Jelly_Star_EEA)
Device model: Jelly Star
```

--- a/ui-weather-view/src/main/java/org/breezyweather/ui/theme/weatherView/materialWeatherView/implementor/SnowImplementor.kt
+++ b/ui-weather-view/src/main/java/org/breezyweather/ui/theme/weatherView/materialWeatherView/implementor/SnowImplementor.kt
@@ -73,10 +73,9 @@ class SnowImplementor(
                 -radius
             }
 
-            val bound = (2 * speedY).toInt()
-            // Fix bound to at least two to prevent crashes on small screens (where bound would be 0) and provide
-            // some variation on medium screens (where bound would always be 1 and the RNG would only give 0)
-            speedX = r.nextInt(if (bound >= 2) bound else 2) - speedY
+            // Fix input to nextInt to at least two to prevent crashes on low res screens (where bound would be 0)
+            // and provide some variation on medium-res screens (where bound would always be 1)
+            speedX = r.nextInt((2 * speedY).toInt().coerceAtLeast(2)) - speedY
             computeCenterPosition()
         }
 

